### PR TITLE
`phone-search` analyzer: don't emit sip/tel prefix, int'l prefix, extension & unformatted input

### PR DESCRIPTION
**Pull request by rursprung**
### Description
see the individual commit messages for further details.

### Related Issues
n/a

### Check List
- [x] Functionality includes testing.
- [x] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable. (no API change => nothing done)
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable. (not docs relevant => nothing done)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by rursprung** [comment]
please add the backport 2.x label (and if there's a chance for a 2.18.1 release please also the backport to that 🙂

**Comment by rursprung** [comment]
> ❕ Gradle check result for [9e120b7](https://github.com/opensearch-project/OpenSearch/commit/9e120b7c7677e9d6ab7e92e916dcc154f9ee73fa): [UNSTABLE](https://build.ci.opensearch.org/job/gradle-check/52053/)
> 
>     * **TEST FAILURES:**
> 
> 
> ```
>       1 org.opensearch.indices.replication.SegmentReplicationIT.testNodeDropWithOngoingReplication
>       1 org.opensearch.action.admin.cluster.node.tasks.ResourceAwareTasksTests.testTaskResourceTrackingDuringTaskCancellation
> ```
> 
> Please review all [flaky tests](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) that succeeded after retry and create an issue if one does not already exist to track the flaky failure.

these are both flaky tests:
* https://github.com/opensearch-project/OpenSearch/issues/14328
* https://github.com/opensearch-project/OpenSearch/issues/14293

**Comment by reta** [comment]
@rursprung could you please resolve the conflicts? thank you

**Comment by rursprung** [comment]
> @rursprung could you please resolve the conflicts? thank you

done.

i've also added a 2nd commit and am now emitting even fewer tokens. please see the commit messages for further details. with the new test coverage i'm now quite confident that this is working as intended

File: CHANGELOG.md [hunk 1]
```diff
@@ -92,6 +92,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Always use `constant_score` query for `match_only_text` field ([#16964](https://github.com/opensearch-project/OpenSearch/pull/16964))
 - Fix Shallow copy snapshot failures on closed index ([#16868](https://github.com/opensearch-project/OpenSearch/pull/16868))
 - Fix multi-value sort for unsigned long ([#16732](https://github.com/opensearch-project/OpenSearch/pull/16732))
+- The `phone-search` analyzer no longer emits the international calling code as a token ([#16993](https://github.com/opensearch-project/OpenSearch/pull/16993))
```
Comment (reta) [inline CHANGELOG.md:95]: Seems like duplicate lines ..

Comment (rursprung) [reply CHANGELOG.md:95]: no, why? these are two different things. i just did them in the same PR since they're modifying the same sources.

**Comment by rursprung** [comment]
> ❌ Gradle check result for [ff3c8da](https://github.com/opensearch-project/OpenSearch/commit/ff3c8dab0843ff652e9d1afb27b5f26da73c3a7e): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/52061/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

flaky test: https://github.com/opensearch-project/OpenSearch/issues/16658

File: CHANGELOG.md (hunk 1 above)
Comment (reta) [reply CHANGELOG.md:95]: Oh I see, got it

**Comment by rursprung** [comment]
> ❌ Gradle check result for [ff3c8da](https://github.com/opensearch-project/OpenSearch/commit/ff3c8dab0843ff652e9d1afb27b5f26da73c3a7e): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/52063/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

flaky test: https://github.com/opensearch-project/OpenSearch/issues/15826

File: CHANGELOG.md (hunk 1 above)
Comment (rursprung) [reply CHANGELOG.md]: i've now merged it into once since i added a 3rd commit and think it's a bit pretentious to have three lines in the changelog for the same analyzer... 😅

**Comment by rursprung** [comment]
> with the new test coverage i'm now quite confident that this is working as intended

aaaand that confidence was misplaced, i spotted another corner case and fixed that as well (incl. increased test coverage). _now_ it should be fine 😅

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
@rursprung could you please backport to 2.x manually? thank you

File: plugins/analysis-phonenumber/src/test/java/org/opensearch/analysis/phone/PhoneNumberAnalyzerTests.java [hunk 2]
```diff
@@ -166,6 +162,10 @@ public void testTelPrefix() throws IOException {
         assertTokensInclude("tel:+1228", Arrays.asList("1228", "122", "228"));
     }
 
+    public void testTelPrefixSearch() throws IOException {
+        assertTokensInclude("tel:+1228", Arrays.asList("1228"));
```
Comment (rursprung) [inline plugins/analysis-phonenumber/src/test/java/org/opensearch/analysis/phone/PhoneNumberAnalyzerTests.java:166]: nit to self: should've been `assertTokensAreInAnyOrder` to make sure that the others aren't present... missed that before 🙈

Comment (rursprung) [reply plugins/analysis-phonenumber/src/test/java/org/opensearch/analysis/phone/PhoneNumberAnalyzerTests.java:166]: done: #17016
